src/TopPage/FluentExtension.php

```diff
@@ -39,10 +39,49 @@ protected function assignTopPage(Page $page): void
         $this->owner->TopPageLocale = FluentState::singleton()->getLocale();
     }
 
+    /*
+     * @inheritdoc
+     */
     protected function clearTopPage(): void
     {
         parent::clearTopPage();
 
         $this->owner->TopPageLocale = null;
     }
+
+    /*
+     * @inheritdoc
+     */
+    protected function assignFixedTopPage(): void
+    {
+        parent::assignFixedTopPage();
+
+        $this->owner->TopPageLocale = FluentState::singleton()->getLocale();
+    }
+
+    /*
+     * @inheritdoc
+     */
+    protected function saveChanges(): void
```

Comment:
There's code duplication going on here so Idon't  think we need this function, instead change the parent function to `DataExtension::saveChanges(array $extraData = [])`

In the parent function in the SQLUpdate have this for the 2nd arg:

```
array_merge(['"TopPageID"' => $owner->TopPageID], $extraData)
```

Reply:
I'll make the change, happy to have the code more DRY.